packages/scroll-sync/src/useMergeRefs.tsx

```diff
@@ -0,0 +1,23 @@
+import { ForwardedRef, MutableRefObject, useMemo } from 'react';
+
+function setRef<T>(ref: ForwardedRef<T>, value: T) {
+	if (typeof ref === 'function') {
+		ref(value);
+	} else if (ref !== null && ref !== undefined) {
+		(ref as MutableRefObject<T>).current = value;
+	}
+}
+
+export function useMergeRefs<T>(...refs: Array<ForwardedRef<T>>): ((refValue: T) => void) | null {
```

Comment:
We should consider extracting this in separate package `@infinum/use-merge-refs`